apiv2/internal/server/os_update_run.go

```diff
@@ -5,31 +5,121 @@ package server
 
 import (
 	"context"
+	"time"
 
-	"google.golang.org/grpc/codes"
+	"google.golang.org/protobuf/types/known/timestamppb"
 
 	computev1 "github.com/open-edge-platform/infra-core/apiv2/v2/internal/pbapi/resources/compute/v1"
+	statusv1 "github.com/open-edge-platform/infra-core/apiv2/v2/internal/pbapi/resources/status/v1"
 	restv1 "github.com/open-edge-platform/infra-core/apiv2/v2/internal/pbapi/services/v1"
+	inv_computev1 "github.com/open-edge-platform/infra-core/inventory/v2/pkg/api/compute/v1"
+	inventory "github.com/open-edge-platform/infra-core/inventory/v2/pkg/api/inventory/v1"
 	"github.com/open-edge-platform/infra-core/inventory/v2/pkg/errors"
+	"github.com/open-edge-platform/infra-core/inventory/v2/pkg/validator"
 )
 
-func (is *InventorygRPCServer) ListOSUpdateRun(_ context.Context, _ *restv1.ListOSUpdateRunRequest) (
+func fromInvOSUpdateRunResource(invOSUpdateRunResource *inv_computev1.OSUpdateRunResource) *computev1.OSUpdateRun {
+
+	parseTimestamp := func(ts string) *timestamppb.Timestamp {
+		parsedTime, err := time.Parse(time.RFC3339, ts)
+		if err != nil {
+			zlog.Warn().Err(err).Msgf("Failed to parse timestamp: %s", ts)
+			return nil
+		}
+		return timestamppb.New(parsedTime)
+	}
```

Comment:
Are you sure about this parsing? I believe we use a different format, see https://github.com/open-edge-platform/infra-core/blob/main/apiv2/internal/server/utils.go#L20